### PR TITLE
chore: auth lint fixes for cdk V2

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
- lint fixes in auth category for cdkV2
-  ;int fixes diabled are:
     - // eslint-disable-next-line spellcheck/spell-checker,
     - /* eslint-disable import/no-cycle */

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.test.ts [hunk 1]
```diff
@@ -9,9 +9,12 @@ import { CognitoStackOptions } from '../../../../provider-utils/awscloudformatio
 
 describe('generateCognitoStackResources', () => {
   const props: CognitoStackOptions = {
+    // eslint-disable-next-line spellcheck/spell-checker
```
Comment (sobolk) [inline packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.test.ts:12]: 1. if possible we should scope it at function level (or perhaps entire file).
2. can you try adding `skipIfMatch` rules that detect hashes? see https://gist.github.com/jrencz/becd1befac3f14b53039e0361087794c#file-eslintrc-L295 as an example.

**Review by sobolk — APPROVED** [review]
LGTM. Please try suggestions before merging.

File: packages/amplify-category-auth/package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "aws-sdk": "^2.1169.0",
     "chalk": "^4.1.1",
     "change-case": "^4.1.1",
+    "cloudform-types": "^4.2.0",
```
[the remark this answers is not in the document]
Comment (akshbhu) [reply packages/amplify-category-auth/package.json:44]: yes

File: packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.test.ts (hunk 1 above)
Comment (akshbhu) [reply packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.test.ts:12]: I didn't find any specific rule that matches all those uuid() + resource names.

I think this is better , enabling at top will hinder any future iterations on this file.

File: packages/amplify-category-auth/package.json (hunk 2 above)
Comment (jhockett) [inline packages/amplify-category-auth/package.json:44]: This is already listed below in `devDependencies`.

Perhaps changing the two times it's imported from 

```ts
import { Template } from 'cloudform-types';
```

to 

```ts
import type { Template } from 'cloudform-types';
```

would resolve the need to add this to `dependencies`.

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts [hunk 3]
```diff
@@ -38,6 +37,7 @@ export class AmplifyAuthTransform extends AmplifyCategoryTransform {
     this._app = new cdk.App();
     this._category = AmplifyCategories.AUTH;
     this._service = AmplifySupportedService.COGNITO;
+    // eslint-disable-next-line spellcheck/spell-checker
     this._authTemplateObj = new AmplifyAuthCognitoStack(this._app, 'AmplifyAuthCongitoStack', { synthesizer: this._synthesizer });
```
[the remark this answers is not in the document]
Comment (akshbhu) [reply packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:41]: I tried this adding to dict , but it always adds with lower case .

**Review by sobolk — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Pull request merged**